packages/app-desktop/gui/Sidebar/hooks/useOnRenderItem.tsx

```diff
@@ -0,0 +1,434 @@
+	}, [props.folders]);
+
+	const selectedIndexRef = useRef(props.selectedIndex);
+	selectedIndexRef.current = props.selectedIndex;
+
+	return useCallback((item: ListItem, index: number) => {
+		const selected = props.selectedIndex === index;
+		const anchorRefCallback = selected ? (
+			(element: HTMLElement) => {
+				if (selectedIndexRef.current === index) {
+					props.onSelectedElementShown(element);
+				}
+			}
+		) : null;
+
+		if (item.kind === ListItemType.Tag) {
+			const tag = item.tag;
+			return <TagItem
+				key={item.key}
+				anchorRef={anchorRefCallback}
+				selected={selected}
+				onClick={tagItem_click}
+				onTagDrop={onTagDrop_}
+				onContextMenu={onItemContextMenu}
+				tag={tag}
+			/>;
+		} else if (item.kind === ListItemType.Folder) {
+			const folder = item.folder;
+			const isExpanded = props.collapsedFolderIds.indexOf(folder.id) < 0;
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
+			let noteCount = (folder as any).note_count;
+
+			// For now hide the count for folders in the trash because it doesn't work and getting it to
+			// work would be tricky.
+			if (folder.deleted_time || folder.id === getTrashFolderId()) noteCount = 0;
+
+			// Thunderbird count: Subtract children note_count from parent folder if it expanded.
+			if (isExpanded) {
+				for (let i = 0; i < props.folders.length; i++) {
+					if (props.folders[i].parent_id === folder.id) {
+						// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
+						noteCount -= (props.folders[i] as any).note_count;
+					}
+				}
+			}
+			return <FolderItem
+				key={item.key}
+				anchorRef={anchorRefCallback}
+				selected={selected}
+				folderId={folder.id}
+				folderTitle={Folder.displayTitle(folder)}
+				folderIcon={Folder.unserializeIcon(folder.icon)}
+				depth={item.depth}
+				isExpanded={isExpanded}
+				hasChildren={item.hasChildren}
+				noteCount={noteCount}
+				onFolderDragStart_={onFolderDragStart_}
+				onFolderDragOver_={onFolderDragOver_}
+				onFolderDrop_={onFolderDrop_}
+				itemContextMenu={onItemContextMenu}
+				folderItem_click={folderItem_click}
+				onFolderToggleClick_={onFolderToggleClick_}
+				shareId={folder.share_id}
+				parentId={folder.parent_id}
+				showFolderIcon={showFolderIcons}
+			/>;
+		} else if (item.kind === ListItemType.Header) {
+			return <HeaderItem
+				key={item.id}
+				item={item}
+				anchorRef={anchorRefCallback}
+				contextMenuHandler={header_contextMenu}
+				onDrop={item.supportsFolderDrop ? onFolderDrop_ : null}
+			/>;
+		} else if (item.kind === ListItemType.AllNotes) {
+			return <AllNotesItem
+				key={item.key}
+				selected={selected}
+				anchorRef={anchorRefCallback}
+			/>;
+		} else if (item.kind === ListItemType.Spacer) {
+			return (
+				<a key={item.key} className='sidebar-spacer-item' ref={anchorRefCallback} aria-label={_('Spacer')}></a>
+			);
```

Comment:
Including a spacer makes the gap between the last notebook and "Tags" much larger than it was before this pull request:

**Screenshot with this pull request**:
<img alt="screenshot: larger space between trash and tags list items" src="https://github.com/laurent22/joplin/assets/46334387/1d5fc4a6-189a-4891-98c0-4fa707548ae4" width="300"/>

**Screenshot with this pull request, with the spacer removed.**
<img src="https://github.com/laurent22/joplin/assets/46334387/a05806ce-09f8-43f4-85c6-efcf49deedb2" width="300"/>

**Screenshot from c6c7de286acef8a12ec136d3d1b59bfcd374c52c (before this pull request)**:
<img src="https://github.com/laurent22/joplin/assets/46334387/b7be36e6-8c4a-4ec4-bdae-7fcb2c72b120" width="300"/>